### PR TITLE
[WFLY-6406] Artemis AIO journal fails on zfsonlinux ZFS filesystem

### PR DESCRIPTION
**Pull request by dpospisil**
JIRA: https://issues.jboss.org/browse/WFLY-6406

Upstream Issue: https://issues.apache.org/jira/browse/ARTEMIS-444
Upstream PR: https://github.com/apache/activemq-artemis/pull/425

Note: the fix relies on component upgrade of activemq-artemis.

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java [hunk 1]
```diff
@@ -161,16 +169,12 @@ public synchronized void start(final StartContext context) throws StartException
                     ROOT_LOGGER.aioInfo();
                 }
                 configuration.setJournalType(JournalType.NIO);
+            } else if (! AIOSequentialFileFactory.isSupported(configuration.getJournalDirectory())) {
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java]: A single check on AIOSequentialFileFactory.isSupported(String) is enough as it encapsulates a call to AIOSequentialFileFactory.isSupported().

Comment (dpospisil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java]: The reason is to issue a different log message if AIO loading fails and if AIO load success but AIO open fails. See https://github.com/apache/activemq-artemis/pull/425

Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java]: Ok, no problem. I'm not sure that it really helps to have 3 different INFO log related to AIO. At the end of the day, "AIO is not supported by the OS file system, switching to NIO" looks to cover all cases :)

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/logging/MessagingLogger.java [hunk 2]
```diff
@@ -815,4 +815,11 @@
 
     @Message(id = 86, value = "Unable to load module %s")
     OperationFailedException unableToLoadModule(String moduleName, @Cause ModuleLoadException cause);
+
+    /**
+     * Logs a info message indicating AIO was not found.
+     */
+    @LogMessage(level = INFO)
+    @Message(id = 87, value = "AIO was found but the filesystem does not support AIO. Switching the configuration into NIO. Journal path: %s")
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/logging/MessagingLogger.java]: For consistency sake, could you word it similarly to `aioInfoLinux()`?

**Comment by dpospisil** [comment]
I have updated the log message to match the other log message. If you still don't like plese update it as you wish or provide me with the exact wording of your preference.

**Comment by dpospisil** [comment]
Also if you want to remove the extra log message please do so I just wanted to mimic the behaviour of the standalone artemis server. Many thanks.

**Comment by n1hility** [comment]
@jmesnil assuming this is rebased and passes CI, wdyt?

**Comment by jmesnil** [comment]
@n1hility -1 this PR requires an upstream change in Artemis that has not been backported to the `-wildfly` branch (PR was rejected in https://github.com/rh-messaging/jboss-activemq-artemis/pull/72 as it was only accepting blocker issues at that time)

**Comment by bstansberry** [comment]
Please open a new PR when the required component upgrade is available.

**Pull request closed without merging**